diff --git a/src/_discord/http.py b/src/_discord/http.py
index 79455a0..4c16c2b 100644
--- a/src/_discord/http.py
+++ b/src/_discord/http.py
@@ -25,6 +25,8 @@ DEALINGS IN THE SOFTWARE.
 
 from __future__ import annotations
 
+import json
+import base64
 import asyncio
 import logging
 import sys
@@ -100,6 +102,22 @@ async def json_or_text(response: aiohttp.ClientResponse) -> dict[str, Any] | str
     return text
 
 
+class UserLimit:
+    def __init__(self, loop: asyncio.BaseEventLoop) -> None:
+        self.usages = 0
+        self.loop = loop
+        self.lock = asyncio.Lock()
+
+    async def ensure(self):
+        await self.lock.acquire()
+        self.usages += 1
+        if self.usages >= 2:
+            self.loop.call_later(4.5, self.lock.release)
+            self.usages = 0
+        else:
+            self.lock.release()
+
+
 class Route:
     def __init__(self, method: str, path: str, **parameters: Any) -> None:
         self.path: str = path
@@ -166,7 +184,7 @@ class HTTPClient:
         proxy: str | None = None,
         proxy_auth: aiohttp.BasicAuth | None = None,
         loop: asyncio.AbstractEventLoop | None = None,
-        unsync_clock: bool = True,
+        unsync_clock: bool = True
     ) -> None:
         self.loop: asyncio.AbstractEventLoop = (
             asyncio.get_event_loop() if loop is None else loop
@@ -177,7 +195,7 @@ class HTTPClient:
         self._global_over: asyncio.Event = asyncio.Event()
         self._global_over.set()
         self.token: str | None = None
-        self.bot_token: bool = False
+        self.bot_token: bool = True
         self.proxy: str | None = proxy
         self.proxy_auth: aiohttp.BasicAuth | None = proxy_auth
         self.use_clock: bool = not unsync_clock
@@ -188,6 +206,7 @@ class HTTPClient:
         self.user_agent: str = user_agent.format(
             __version__, sys.version_info, aiohttp.__version__
         )
+        self.user_limit = UserLimit(self.loop)
 
     def recreate(self) -> None:
         if self.__session.closed:
@@ -234,8 +253,23 @@ class HTTPClient:
             "User-Agent": self.user_agent,
         }
 
+        # Self-bot modification
+        if not self.bot_token:
+            headers["x-super-properties"] = base64.b64encode(
+                json.dumps(
+                    {
+                        "os": sys.platform,
+                        "browser":"Chrome",
+                        "system_locale":"en-US",
+                        "browser_user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64)",
+                        "browser_version":"109.0.0.0",
+                    },
+                ).encode("utf-8")
+            ).decode("utf-8")
+            headers["origin"] = "https://discord.com"
+
         if self.token is not None:
-            headers["Authorization"] = f"Bot {self.token}"
+            headers["Authorization"] = self.token
         # some checking if it's a JSON request
         if "json" in kwargs:
             headers["Content-Type"] = "application/json"
@@ -279,6 +313,11 @@ class HTTPClient:
                     kwargs["data"] = form_data
 
                 try:
+
+                    if not self.bot_token:
+                        await self.user_limit.ensure()
+
+
                     async with self.__session.request(
                         method, url, **kwargs
                     ) as response:
@@ -316,7 +355,7 @@ class HTTPClient:
 
                         # we are being rate limited
                         if response.status == 429:
-                            if not response.headers.get("Via") or isinstance(data, str):
+                            if not response.headers.get("Via") or isinstance(data, str) or ("code" in data and data["code"] == 20016):
                                 # Banned by Cloudflare more than likely.
                                 raise HTTPException(response, data)
 
@@ -401,22 +440,28 @@ class HTTPClient:
 
     # login management
 
-    async def static_login(self, token: str) -> user.User:
+    async def static_login(self, token: str, bot: bool) -> user.User:
         # Necessary to get aiohttp to stop complaining about session creation
         self.__session = aiohttp.ClientSession(
             connector=self.connector, ws_response_class=DiscordClientWebSocketResponse
         )
         old_token = self.token
-        self.token = token
+        old_bot = self.bot_token
+        self.token = f"Bot {token}" if bot else token
+        self.bot_token = bot
 
         try:
             data = await self.request(Route("GET", "/users/@me"))
         except HTTPException as exc:
             self.token = old_token
+            self.bot_token = old_bot
             if exc.status == 401:
                 raise LoginFailure("Improper token has been passed.") from exc
             raise
 
+        if not bot:
+            self.user_agent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
+
         return data
 
     def logout(self) -> Response[None]:
