diff --git a/src/_discord/http.py b/src/_discord/http.py
index a131e59..65c46c1 100644
--- a/src/_discord/http.py
+++ b/src/_discord/http.py
@@ -25,6 +25,8 @@ DEALINGS IN THE SOFTWARE.
 
 from __future__ import annotations
 
+import json
+import base64
 import asyncio
 import logging
 import sys
@@ -115,6 +117,22 @@ async def json_or_text(response: aiohttp.ClientResponse) -> dict[str, Any] | str
     return text
 
 
+class UserLimit:
+    def __init__(self, loop: asyncio.BaseEventLoop) -> None:
+        self.usages = 0
+        self.loop = loop
+        self.lock = asyncio.Lock()
+
+    async def ensure(self):
+        await self.lock.acquire()
+        self.usages += 1
+        if self.usages >= 2:
+            self.loop.call_later(4.5, self.lock.release)
+            self.usages = 0
+        else:
+            self.lock.release()
+
+
 class Route:
     API_BASE_URL: str = "https://discord.com/api/v{API_VERSION}"
 
@@ -183,7 +201,7 @@ class HTTPClient:
         proxy: str | None = None,
         proxy_auth: aiohttp.BasicAuth | None = None,
         loop: asyncio.AbstractEventLoop | None = None,
-        unsync_clock: bool = True,
+        unsync_clock: bool = True
     ) -> None:
         self.loop: asyncio.AbstractEventLoop = (
             asyncio.get_event_loop() if loop is None else loop
@@ -194,7 +212,7 @@ class HTTPClient:
         self._global_over: asyncio.Event = asyncio.Event()
         self._global_over.set()
         self.token: str | None = None
-        self.bot_token: bool = False
+        self.bot_token: bool = True
         self.proxy: str | None = proxy
         self.proxy_auth: aiohttp.BasicAuth | None = proxy_auth
         self.use_clock: bool = not unsync_clock
@@ -205,6 +223,7 @@ class HTTPClient:
         self.user_agent: str = user_agent.format(
             __version__, sys.version_info, aiohttp.__version__
         )
+        self.user_limit = UserLimit(self.loop)
 
     def recreate(self) -> None:
         if self.__session.closed:
@@ -251,8 +270,23 @@ class HTTPClient:
             "User-Agent": self.user_agent,
         }
 
+        # Self-bot modification
+        if not self.bot_token:
+            headers["x-super-properties"] = base64.b64encode(
+                json.dumps(
+                    {
+                        "os": sys.platform,
+                        "browser":"Chrome",
+                        "system_locale":"en-US",
+                        "browser_user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64)",
+                        "browser_version":"109.0.0.0",
+                    },
+                ).encode("utf-8")
+            ).decode("utf-8")
+            headers["origin"] = "https://discord.com"
+
         if self.token is not None:
-            headers["Authorization"] = f"Bot {self.token}"
+            headers["Authorization"] = self.token
         # some checking if it's a JSON request
         if "json" in kwargs:
             headers["Content-Type"] = "application/json"
@@ -296,6 +330,11 @@ class HTTPClient:
                     kwargs["data"] = form_data
 
                 try:
+
+                    if not self.bot_token:
+                        await self.user_limit.ensure()
+
+
                     async with self.__session.request(
                         method, url, **kwargs
                     ) as response:
@@ -335,7 +374,8 @@ class HTTPClient:
 
                         # we are being rate limited
                         if response.status == 429:
-                            if not response.headers.get("Via") or isinstance(data, str):
+                            retry_after: float = data.get("retry_after", 0)
+                            if not response.headers.get("Via") or isinstance(data, str) or ("code" in data and data["code"] == 20016) or retry_after > 30:
                                 # Banned by Cloudflare more than likely.
                                 raise HTTPException(response, data)
 
@@ -345,7 +385,6 @@ class HTTPClient:
                             )
 
                             # sleep a bit
-                            retry_after: float = data["retry_after"]
                             _log.warning(fmt, retry_after, bucket)
 
                             # check if it's a global rate limit
@@ -437,13 +476,17 @@ class HTTPClient:
 
     # login management
 
-    async def static_login(self, token: str) -> user.User:
+    async def static_login(self, token: str, bot: bool = True) -> user.User:
         # Necessary to get aiohttp to stop complaining about session creation
         self.__session = aiohttp.ClientSession(
             connector=self.connector, ws_response_class=DiscordClientWebSocketResponse
         )
         old_token = self.token
-        self.token = token
+        self.bot_token = bot
+        self.token = f"Bot {token}" if bot else token
+
+        if not bot:
+            self.user_agent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
 
         try:
             data = await self.request(Route("GET", "/users/@me"))
