diff --git a/src/_discord/client.py b/src/_discord/client.py
index 584fa35..b96c914 100644
--- a/src/_discord/client.py
+++ b/src/_discord/client.py
@@ -122,7 +122,7 @@ def _cleanup_loop(loop: asyncio.AbstractEventLoop) -> None:
         loop.run_until_complete(loop.shutdown_asyncgens())
     finally:
         _log.info("Closing the event loop.")
-        loop.close()
+        loop.call_soon(loop.close)
 
 
 class Client:
@@ -221,6 +221,7 @@ class Client:
         self,
         *,
         loop: Optional[asyncio.AbstractEventLoop] = None,
+        bot: Optional[bool] = True,
         **options: Any,
     ):
         # self.ws is set in the connect method
@@ -240,6 +241,7 @@ class Client:
             proxy_auth=proxy_auth,
             unsync_clock=unsync_clock,
             loop=self.loop,
+            bot=bot
         )
 
         self._handlers: Dict[str, Callable] = {"ready": self._handle_ready}
@@ -1563,6 +1565,10 @@ class Client:
         """
         data = await self.http.get_user(user_id)
         return User(state=self._connection, data=data)
+    
+    async def fetch_relationships(self) -> List[User]:
+        data = await self.http.get_relationships()
+        return [User(state=self._connection, data=d["user"]) for d in data]
 
     async def fetch_channel(self, channel_id: int, /) -> Union[GuildChannel, PrivateChannel, Thread]:
         """|coro|
