diff --git a/src/_discord/gateway.py b/src/_discord/gateway.py
index e3e7d1b..e847fc6 100644
--- a/src/_discord/gateway.py
+++ b/src/_discord/gateway.py
@@ -302,6 +302,7 @@ class DiscordWebSocket:
         self._buffer = bytearray()
         self._close_code = None
         self._rate_limiter = GatewayRatelimiter()
+        self.bot: bool = True
 
     @property
     def open(self):
@@ -338,6 +339,7 @@ class DiscordWebSocket:
 
         # dynamically add attributes needed
         ws.token = client.http.token
+        ws.bot = client.http.bot_token
         ws._connection = client._connection
         ws._discord_parsers = client._connection.parsers
         ws._dispatch = client.dispatch
@@ -398,23 +400,41 @@ class DiscordWebSocket:
 
     async def identify(self):
         """Sends the IDENTIFY packet."""
-        payload = {
-            "op": self.IDENTIFY,
-            "d": {
-                "token": self.token,
-                "properties": {
-                    "os": sys.platform,
-                    "browser": "pycord",
-                    "device": "pycord",
+        if self.bot:
+            payload = {
+                "op": self.IDENTIFY,
+                "d": {
+                    "token": self.token,
+                    "properties": {
+                        "os": sys.platform,
+                        "browser": "pycord",
+                        "device": "pycord",
+                    },
+                    "compress": True,
+                    "large_threshold": 250,
+                    "v": 3,
                 },
-                "compress": True,
-                "large_threshold": 250,
-                "v": 3,
-            },
-        }
+            }
 
-        if self.shard_id is not None and self.shard_count is not None:
-            payload["d"]["shard"] = [self.shard_id, self.shard_count]
+            if self.shard_id is not None and self.shard_count is not None:
+                payload["d"]["shard"] = [self.shard_id, self.shard_count]
+        
+        else:
+            payload = {
+                "op": self.IDENTIFY,
+                "d": {
+                    "token": self.token,
+                    "properties": {
+                        "os": sys.platform,
+                        "browser":"Chrome",
+                        "system_locale":"en-US",
+                        "browser_user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64)",
+                        "browser_version":"109.0.0.0",
+                    },
+                    "presence": {"status":"online","since":0,"activities":[],"afk": False},
+                    "compress": True
+                }
+            }
 
         state = self._connection
         if state._activity is not None or state._status is not None:
