diff --git a/src/_discord/_version.py b/src/_discord/_version.py
index c58c8ba..3761f0f 100644
--- a/src/_discord/_version.py
+++ b/src/_discord/_version.py
@@ -37,25 +37,7 @@ from typing import Literal, NamedTuple
 
 from .utils import deprecated
 
-try:
-    __version__ = version("py-cord")
-except PackageNotFoundError:
-    # Package is not installed
-    try:
-        from setuptools_scm import get_version  # type: ignore[import]
-
-        __version__ = get_version()
-    except ImportError:
-        # setuptools_scm is not installed
-        __version__ = "0.0.0"
-        warnings.warn(
-            (
-                "Package is not installed, and setuptools_scm is not installed. "
-                f"As a fallback, {__name__}.__version__ will be set to {__version__}"
-            ),
-            RuntimeWarning,
-            stacklevel=2,
-        )
+__version__ = "2.4.1"
 
 
 class AdvancedVersionInfo(TypedDict):
