diff --git a/src/_discord/__main__.py b/src/_discord/__main__.py
index ed34bdf..9f31165 100644
--- a/src/_discord/__main__.py
+++ b/src/_discord/__main__.py
@@ -32,7 +32,7 @@ from typing import Tuple
 
 import aiohttp
 
-import discord
+import _discord as discord
 
 
 def show_version() -> None:
@@ -64,8 +64,8 @@ def core(parser, args) -> None:
 
 _bot_template = """#!/usr/bin/env python3
 
-from discord.ext import commands
-import discord
+from _discord.ext import commands
+import _discord as discord
 import config
 
 class Bot(commands.{base}):
@@ -118,8 +118,8 @@ var/
 config.py
 """
 
-_cog_template = '''from discord.ext import commands
-import discord
+_cog_template = '''from _discord.ext import commands
+import _discord as discord
 
 class {name}(commands.Cog{attrs}):
     """The description for {name} goes here."""
diff --git a/src/_discord/channel.py b/src/_discord/channel.py
index 76d6f6e..b5a3902 100644
--- a/src/_discord/channel.py
+++ b/src/_discord/channel.py
@@ -38,7 +38,7 @@ from typing import (
     overload,
 )
 
-import discord.abc
+import _discord as discord
 
 from . import utils
 from .asset import Asset
diff --git a/src/_discord/cog.py b/src/_discord/cog.py
index 5fbe6e6..4f5039b 100644
--- a/src/_discord/cog.py
+++ b/src/_discord/cog.py
@@ -43,7 +43,7 @@ from typing import (
 
 from typing_extensions import TypeGuard
 
-import discord.utils
+import _discord as discord
 
 from . import errors
 from .commands import (
diff --git a/src/_discord/commands/context.py b/src/_discord/commands/context.py
index b9d6f8e..a310c50 100644
--- a/src/_discord/commands/context.py
+++ b/src/_discord/commands/context.py
@@ -27,9 +27,9 @@ from __future__ import annotations
 
 from typing import TYPE_CHECKING, Any, TypeVar, overload
 
-import discord.abc
-from discord.interactions import Interaction, InteractionMessage, InteractionResponse
-from discord.webhook.async_ import Webhook
+import _discord as discord
+from _discord.interactions import Interaction, InteractionMessage, InteractionResponse
+from _discord.webhook.async_ import Webhook
 
 if TYPE_CHECKING:
     from typing import Awaitable, Callable
diff --git a/src/_discord/ext/bridge/bot.py b/src/_discord/ext/bridge/bot.py
index 9e2c619..12071d3 100644
--- a/src/_discord/ext/bridge/bot.py
+++ b/src/_discord/ext/bridge/bot.py
@@ -28,10 +28,10 @@ from __future__ import annotations
 from abc import ABC
 from collections.abc import Iterator
 
-from discord.commands import ApplicationContext
-from discord.errors import CheckFailure, DiscordException
-from discord.interactions import Interaction
-from discord.message import Message
+from _discord.commands import ApplicationContext
+from _discord.errors import CheckFailure, DiscordException
+from _discord.interactions import Interaction
+from _discord.message import Message
 
 from ..commands import AutoShardedBot as ExtAutoShardedBot
 from ..commands import Bot as ExtBot
diff --git a/src/_discord/ext/bridge/context.py b/src/_discord/ext/bridge/context.py
index 8e7f941..f771b5f 100644
--- a/src/_discord/ext/bridge/context.py
+++ b/src/_discord/ext/bridge/context.py
@@ -28,10 +28,10 @@ from __future__ import annotations
 from abc import ABC, abstractmethod
 from typing import TYPE_CHECKING, Any, Union, overload
 
-from discord.commands import ApplicationContext
-from discord.interactions import Interaction, InteractionMessage
-from discord.message import Message
-from discord.webhook import WebhookMessage
+from _discord.commands import ApplicationContext
+from _discord.interactions import Interaction, InteractionMessage
+from _discord.message import Message
+from _discord.webhook import WebhookMessage
 
 from ..commands import Context
 
diff --git a/src/_discord/ext/bridge/core.py b/src/_discord/ext/bridge/core.py
index e61daa5..31aa6ec 100644
--- a/src/_discord/ext/bridge/core.py
+++ b/src/_discord/ext/bridge/core.py
@@ -29,7 +29,7 @@ import inspect
 from collections.abc import Iterator
 from typing import TYPE_CHECKING, Any, Callable
 
-import discord.commands.options
+import _discord.commands.options
 from discord import (
     ApplicationCommand,
     Attachment,
diff --git a/src/_discord/ext/commands/bot.py b/src/_discord/ext/commands/bot.py
index d348b6c..0e750cf 100644
--- a/src/_discord/ext/commands/bot.py
+++ b/src/_discord/ext/commands/bot.py
@@ -31,7 +31,7 @@ import sys
 import traceback
 from typing import TYPE_CHECKING, Any, Callable, Coroutine, Iterable, TypeVar
 
-import discord
+import _discord as discord
 
 from . import errors
 from .context import Context
diff --git a/src/_discord/ext/commands/cog.py b/src/_discord/ext/commands/cog.py
index 871d3e8..1593518 100644
--- a/src/_discord/ext/commands/cog.py
+++ b/src/_discord/ext/commands/cog.py
@@ -27,7 +27,7 @@ from __future__ import annotations
 
 from typing import TYPE_CHECKING, Any, Callable, Generator, TypeVar
 
-import discord
+import _discord as discord
 
 from ...cog import Cog
 from ...commands import ApplicationCommand, SlashCommandGroup
diff --git a/src/_discord/ext/commands/context.py b/src/_discord/ext/commands/context.py
index afd023d..5bc6b9c 100644
--- a/src/_discord/ext/commands/context.py
+++ b/src/_discord/ext/commands/context.py
@@ -29,9 +29,8 @@ import inspect
 import re
 from typing import TYPE_CHECKING, Any, Generic, TypeVar, Union
 
-import discord.abc
-import discord.utils
-from discord.message import Message
+import _discord as discord
+from _discord.message import Message
 
 if TYPE_CHECKING:
     from typing_extensions import ParamSpec
diff --git a/src/_discord/ext/commands/converter.py b/src/_discord/ext/commands/converter.py
index ca5a109..53c1246 100644
--- a/src/_discord/ext/commands/converter.py
+++ b/src/_discord/ext/commands/converter.py
@@ -40,8 +40,8 @@ from typing import (
     runtime_checkable,
 )
 
-import discord
-from discord.utils import UNICODE_EMOJIS
+import _discord as discord
+from _discord.utils import UNICODE_EMOJIS
 
 from .errors import *
 
diff --git a/src/_discord/ext/commands/cooldowns.py b/src/_discord/ext/commands/cooldowns.py
index 6e58d37..44e3dc4 100644
--- a/src/_discord/ext/commands/cooldowns.py
+++ b/src/_discord/ext/commands/cooldowns.py
@@ -30,8 +30,8 @@ import time
 from collections import deque
 from typing import TYPE_CHECKING, Any, Callable, Deque, TypeVar
 
-import discord.abc
-from discord.enums import Enum
+import _discord.abc
+from _discord.enums import Enum
 
 from ...abc import PrivateChannel
 from .errors import MaxConcurrencyReached
diff --git a/src/_discord/ext/commands/core.py b/src/_discord/ext/commands/core.py
index 4a58ecf..8633f95 100644
--- a/src/_discord/ext/commands/core.py
+++ b/src/_discord/ext/commands/core.py
@@ -42,7 +42,7 @@ from typing import (
     overload,
 )
 
-import discord
+import _discord as discord
 
 from ...commands import (
     ApplicationCommand,
diff --git a/src/_discord/ext/commands/errors.py b/src/_discord/ext/commands/errors.py
index c768fd7..11a6cc1 100644
--- a/src/_discord/ext/commands/errors.py
+++ b/src/_discord/ext/commands/errors.py
@@ -27,7 +27,7 @@ from __future__ import annotations
 
 from typing import TYPE_CHECKING, Any, Callable
 
-from discord.errors import ClientException, DiscordException
+from _discord.errors import ClientException, DiscordException
 
 if TYPE_CHECKING:
     from inspect import Parameter
diff --git a/src/_discord/ext/commands/flags.py b/src/_discord/ext/commands/flags.py
index ec218b8..f7888f6 100644
--- a/src/_discord/ext/commands/flags.py
+++ b/src/_discord/ext/commands/flags.py
@@ -31,7 +31,7 @@ import sys
 from dataclasses import dataclass, field
 from typing import TYPE_CHECKING, Any, Iterator, Literal, Pattern, TypeVar, Union
 
-from discord.utils import (
+from _discord.utils import (
     MISSING,
     maybe_coroutine,
     resolve_annotation,
diff --git a/src/_discord/ext/commands/help.py b/src/_discord/ext/commands/help.py
index 75dc686..955543b 100644
--- a/src/_discord/ext/commands/help.py
+++ b/src/_discord/ext/commands/help.py
@@ -31,7 +31,7 @@ import itertools
 import re
 from typing import TYPE_CHECKING, Any
 
-import discord.utils
+import _discord.utils
 
 from .core import Command, Group
 from .errors import CommandError
diff --git a/src/_discord/ext/pages/pagination.py b/src/_discord/ext/pages/pagination.py
index a35256b..a4ae9f2 100644
--- a/src/_discord/ext/pages/pagination.py
+++ b/src/_discord/ext/pages/pagination.py
@@ -27,13 +27,13 @@ from __future__ import annotations
 import contextlib
 from typing import List
 
-import discord
-from discord.errors import DiscordException
-from discord.ext.bridge import BridgeContext
-from discord.ext.commands import Context
-from discord.file import File
-from discord.member import Member
-from discord.user import User
+import _discord as discord
+from _discord.errors import DiscordException
+from _discord.ext.bridge import BridgeContext
+from _discord.ext.commands import Context
+from _discord.file import File
+from _discord.member import Member
+from _discord.user import User
 
 __all__ = (
     "PaginatorButton",
diff --git a/src/_discord/ext/tasks/__init__.py b/src/_discord/ext/tasks/__init__.py
index 9bdde87..9980ed4 100644
--- a/src/_discord/ext/tasks/__init__.py
+++ b/src/_discord/ext/tasks/__init__.py
@@ -36,9 +36,9 @@ from typing import Any, Awaitable, Callable, Generic, TypeVar, cast
 
 import aiohttp
 
-import discord
-from discord.backoff import ExponentialBackoff
-from discord.utils import MISSING
+import _discord as discord
+from _discord.backoff import ExponentialBackoff
+from _discord.utils import MISSING
 
 __all__ = ("loop",)
 
diff --git a/src/_discord/member.py b/src/_discord/member.py
index 8210d12..1770b82 100644
--- a/src/_discord/member.py
+++ b/src/_discord/member.py
@@ -32,7 +32,7 @@ import sys
 from operator import attrgetter
 from typing import TYPE_CHECKING, Any, TypeVar, Union
 
-import discord.abc
+import _discord as discord
 
 from . import utils
 from .activity import ActivityTypes, create_activity
diff --git a/src/_discord/sinks/errors.py b/src/_discord/sinks/errors.py
index 5f036ef..5fccdf8 100644
--- a/src/_discord/sinks/errors.py
+++ b/src/_discord/sinks/errors.py
@@ -22,7 +22,7 @@ FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 DEALINGS IN THE SOFTWARE.
 """
 
-from discord.errors import DiscordException
+from _discord.errors import DiscordException
 
 
 class SinkException(DiscordException):
diff --git a/src/_discord/types/soundboard.py b/src/_discord/types/soundboard.py
index 9a4c19b..2a80ef9 100644
--- a/src/_discord/types/soundboard.py
+++ b/src/_discord/types/soundboard.py
@@ -26,7 +26,7 @@ from __future__ import annotations
 
 from typing_extensions import NotRequired, TypedDict
 
-from discord.types.user import User
+from _discord.types.user import User
 
 from .snowflake import Snowflake
 
diff --git a/src/_discord/user.py b/src/_discord/user.py
index 0c57392..7090dfb 100644
--- a/src/_discord/user.py
+++ b/src/_discord/user.py
@@ -27,7 +27,7 @@ from __future__ import annotations
 
 from typing import TYPE_CHECKING, Any, TypeVar
 
-import discord.abc
+import _discord as discord
 
 from .asset import Asset
 from .collectibles import Nameplate
