diff --git a/discord/http.py b/discord/http.py
index 3f63a75d..19fe6959 100644
--- a/discord/http.py
+++ b/discord/http.py
@@ -211,33 +211,9 @@ class HTTPClient:
 
                         # we are being rate limited
                         if r.status == 429:
-                            if not r.headers.get('Via'):
-                                # Banned by Cloudflare more than likely.
-                                raise HTTPException(r, data)
-
-                            fmt = 'We are being rate limited. Retrying in %.2f seconds. Handled under the bucket "%s"'
-
-                            # sleep a bit
-                            retry_after = data['retry_after'] / 1000.0
-                            log.warning(fmt, retry_after, bucket)
-
-                            # check if it's a global rate limit
-                            is_global = data.get('global', False)
-                            if is_global:
-                                log.warning('Global rate limit has been hit. Retrying in %.2f seconds.', retry_after)
-                                self._global_over.clear()
-
-                            await asyncio.sleep(retry_after)
-                            log.debug('Done sleeping for the rate limit. Retrying...')
-
-                            # release the global lock now that the
-                            # global rate limit has passed
-                            if is_global:
-                                self._global_over.set()
-                                log.debug('Global rate limit is now over.')
-
-                            continue
+                            raise HTTPException(r, data)
 
+                        
                         # we've received a 500 or 502, unconditional retry
                         if r.status in {500, 502}:
                             await asyncio.sleep(1 + tries * 2)
